x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -291,25 +292,25 @@ private void storeTrainedModelAndDefinition(TrainedModelConfig trainedModelConfi
 
         List<TrainedModelDefinitionDoc> trainedModelDefinitionDocs = new ArrayList<>();
         try {
-            String compressedString = trainedModelConfig.getCompressedDefinition();
-            if (compressedString.length() > MAX_COMPRESSED_STRING_SIZE) {
+            BytesReference compressedDefinition = trainedModelConfig.getCompressedDefinition();
+            if (compressedDefinition.length() > MAX_COMPRESSED_STRING_SIZE) {
                 listener.onFailure(
                     ExceptionsHelper.badRequestException(
                         "Unable to store model as compressed definition has length [{}] the limit is [{}]",
```

Comment:
Might be good to indicate that the length is in UTF-8 bytes.

Reply:
I added the fact the size is in bytes to the message in 7a59661575ee9b138356be4c3087889f7c03b5c8